iroh-dns-server/src/store.rs

```diff
@@ -103,7 +103,8 @@ impl ZoneStore {
         record_type: RecordType,
     ) -> Result<Option<Arc<RecordSet>>> {
         trace!("store resolve");
-        if let Some(rset) = self.cache.lock().await.resolve(pubkey, name, record_type) {
+        let mut cache = self.cache.lock().await;
```

Comment:
This will now hold a lock on the cache until the end of the function, which can include a DHT resolve which can take long. Whereas previously the lock would only be held shortly. If you do want to keep a lock around the guaranteed-to-be-fast operations, then I propose to do a block scopes (`{ ... }`) that makes it clear for how long the lock is held.